### ISSUE TITLE
Support for JSON:API

### ISSUE THREAD

**Issue by manhluong** [flutterchina/dio#632]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Hi,

Are there any plans to support JSON:API? I mean this: https://jsonapi.org/

Maybe with a plugin ?

I tried to look around but could not find anything.

Something similar for retrofit does exist: https://github.com/jasminb/jsonapi-converter

Thanks!
